torchinductor/lowering.py

```diff
@@ -2618,15 +2618,15 @@ def fn(idx):
                     scale,
                 )
 
+                mask = ops.and_(
+                    ops.lt(ph, phend),
+                    ops.lt(pw, pwend),
+                )
                 if gradient is None:
-                    # don't need mask for 0, 0
```

Comment:
@jansel , I had to change this to pass the added test. I wonder if `max_pool2d_with_indices_backward ` has the same issue, https://github.com/pytorch/torchdynamo/blob/138eb96c36c6ba5a14e65d28fad3cbd5a6e6dcfc/torchinductor/lowering.py#L2224, although I haven't tried to construct a test case for that.